### DISCUSSION TITLE
How to log all `selectedKeys` instead of 'all' in the Collections

### DISCUSSION THREAD

**Discussion by sadeghbarati** [adobe/react-spectrum#7182] [Q&A] [answered]
Please check this link as an example https://getjustd.com/docs/components/collections/tree

I hit the <kbd>⌘ Cmd + A</kbd>, How do I get all `selectedKeys` values instead of the `all` keyword?

Also how to disable the `select all` and `esc` keyboard shortcuts?

---

There is some strange behavior that if I have 10 items and I select them all manually with clicks I will have selectedKeys with size of 10 
But if I select them with <kbd>⌘ Cmd + A</kbd> I will have `all` as selectedKeys

**Comment by snowystinger** [comment, marked as answer]
Thanks for the discussion. The special case for 'all' is to handle async collections of items where not all the keys are known up front.
If you would like to convert the 'all' to a set of keys, you can run a loop over your item collections and pull out the keys of each item.

For more information, see https://react-spectrum.adobe.com/react-stately/selection.html#select-all
